apicapi/tests/unit/test_apic_manager.py

```diff
@@ -313,6 +313,24 @@ def test_ensure_infra_created_l3ext_domain(self):
             mocked.APIC_EXT_PORT,
             mgr.apic.infraAccPortGrp.mo.dn(mocked.APIC_L3EXT_FUNC_PROF))
 
+    def test_no_provision_infra(self):
+        self._initialize_manager(
+            phys_domains={mocked.APIC_DOMAIN: {}},
+            vmm_domains={mocked.APIC_DOMAIN + '1': {
+                            'apic_vmm_type': 'OpenStack'},
+                         mocked.APIC_DOMAIN + '2': {
+                             'apic_vmm_type': 'VMware'}})
+        self.mgr.provision_infra = False
+        self.mgr.ensure_entity_profile_created_on_apic = mock.Mock()
+        apic_domain.VmDomain._ensure_vmm_domain_created_on_apic = mock.Mock()
```

Comment:
Please use:

`with mock.patch.object(apic_domain.VmDomainm '_ensure_vmm_domain_created_on_apic'):`

Sometimes, depending on the order in which the tests are run, this mocks don't get properly restored and tests will randomly fail. By mocking a class method using the with statement restoration will happen automatically at the exit of the context.